src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -6003,6 +6003,25 @@ void AuraEffect::HandleStoreTeleportReturnPoint(AuraApplication const* aurApp, u
         playerTarget->RemoveStoredAuraTeleportLocation(GetSpellInfo()->Id);
 }
 
+void AuraEffect::HandleCosmeticMounted(AuraApplication const* aurApp, uint8 mode, bool apply) const
+{
+    if (!(mode & AURA_EFFECT_HANDLE_REAL))
+        return;
+
+    if (apply)
+    {
+        aurApp->GetTarget()->Dismount();
```

Comment:
Not needed here, lack of `SPELL_ATTR0_CASTABLE_WHILE_MOUNTED` is supposed to take care of it